Huffmann/HuffmannTree.lean

```diff
@@ -4,50 +4,57 @@ We will initially define the binary treee for encoding and get the minimum encod
 -/
 
 abbrev BoolList := List Bool
-abbrev AlphaNumList := List (Char × Nat)
-abbrev BoolEncList := List (Char × BoolList)
-abbrev CharEncodedList := List (Char × String)
+abbrev AlphaNumList (α : Type) := List (α × Nat)
+abbrev BoolEncList (α : Type) := List (α × BoolList)
+abbrev TypeEncodedList (α : Type) := List (α × String)
 
-def eg₁ : AlphaNumList := [('a', 45),('b', 13),('c', 12),('d', 16),('e', 9),('f', 5)]
+-- This is made so that I can use for any type of data which is HuffmanType, etc. like Char, Int, etc.
+class HfmnType (α : Type) where
```

Comment:
Use `class HfmnType (α : Type) [DecidableEq α]` and remove the instance as a field.